src/zenml/stack/registry.py

```diff
@@ -0,0 +1,150 @@
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+from collections import defaultdict
+from typing import DefaultDict, Dict, List
+
+from zenml.enums import StackComponentType
+from zenml.logger import get_logger
+from zenml.zen_stores.models import FlavorWrapper
+
+logger = get_logger(__name__)
+
+
+class FlavorRegistry:
+    """Registry for stack component flavor.
+
+    The flavors defined by ZenML must be registered here.
+    """
+
+    def __init__(self) -> None:
+        """Initialization of the flavors."""
+        self._flavors: DefaultDict[
+            StackComponentType, Dict[str, FlavorWrapper]
+        ] = defaultdict(dict)
+
+        self.register_default_flavors()
+        self.register_integration_flavors()
+
+    def register_default_flavors(self) -> None:
+        """Registers the default built-in flavors."""
+        from zenml.artifact_stores import LocalArtifactStore
+        from zenml.container_registries import BaseContainerRegistry
+        from zenml.metadata_stores import (
+            MySQLMetadataStore,
+            SQLiteMetadataStore,
+        )
+        from zenml.orchestrators import LocalOrchestrator
+        from zenml.secrets_managers import LocalSecretsManager
+
+        default_flavors = [
+            LocalOrchestrator,
+            SQLiteMetadataStore,
+            MySQLMetadataStore,
+            LocalArtifactStore,
+            BaseContainerRegistry,
+            LocalSecretsManager,
+        ]
+        for flavor in default_flavors:
+            self._register_flavor(
+                FlavorWrapper(
+                    name=flavor.FLAVOR,  # type: ignore[attr-defined]
+                    type=flavor.TYPE,  # type: ignore[attr-defined]
+                    source=flavor.__module__ + "." + flavor.__name__,
+                    integration="built-in",
+                )
+            )
+
+    def register_integration_flavors(self) -> None:
+        """Registers the flavors implemented by integrations."""
+        from zenml.integrations.registry import integration_registry
+
+        for integration in integration_registry.integrations.values():
+            integrated_flavors = integration.declare()
+            if integrated_flavors:
+                for flavor in integrated_flavors:
+                    self._register_flavor(flavor)
+
+    def _register_flavor(
+        self,
+        flavor: FlavorWrapper,
+    ) -> None:
+        """Registers a stack component flavor."""
+        flavors = self._flavors[flavor.type]
+
+        if flavor.name in flavors:
+            raise KeyError(
+                f"There is already a {flavor.type} with the flavor "
+                f"`{flavor.name}`. Please select another name for the flavor."
+            )
+
+        flavors[flavor.name] = flavor
+        logger.debug(
+            f"Registered flavor for '{flavor.name}' and type '{flavor.type}'.",
+        )
+
+    def get_flavor(
```

Comment:
If the flavor registry is just a "layer" on top of the ZenStore, you could make this even more obvious by defining an abstract class with methods like `get_flavor` and `register_flavor` that both FlavorRegistry and ZenStore inherit from.

I used this same approach with Repository and GlobalConfiguration, given that Repository is kind of a "shim" layer on top of the global configuration when it comes to setting the active profile and active stack.

Reply:
Tried to mention this in the big comment down below. For the time being, I opted to just define a method within the `Repository` to act as a bridge between these two entities. If it is alright, I would keep this idea as a future improvement.